docs/explanation/core/configuration.md

```diff
@@ -656,3 +658,12 @@ being available and must move on if such is not responding.
 
 The impact shown above is negligible in a decentralized network with enough
 decentralization.
+
+### The adverse effect of using inconsistent `timeout_commit` in a network
+
+Let's look at the same scenario as before. There are ten geographically
+distributed validators. One of them (Bob) sets `timeout_commit` to 0s. Others
+have it set to 1s (the default value). Now, Bob will be the fastest producer
+because he doesn't wait for additional precommits after creating a block. If
+waiting for precommits (`timeout_commit`) is not incentivized, Bob will accrue
+more rewards compared to the other 9 validators.
```

Comment:
Not sure about the last sentence. What can happen does not wait for `timeout_commit` (i.e., it is zero):
1. Not all precommits are collected, so some nodes are not rewarded for them
1. Bob starts `timeout_propose` before all the others, therefore will stop waiting for other nodes' blocks early, this possibly prevoting `nil` for blocks other will accept an prevote for.

To achieve more rewards, Bob has to consider its vote in case 1. and ignore others. But it has in any case to include 2/3+ of them.

In case 2., if Bob has the same voting power than the others, or, more precisely, if it has `<1/3` of the voting power, the fact it prevotes for `nil` should not have impact, as other nodes will accept blocks proposed by other nodes.

Also regarding 1., Bob has the advantage of broadcasting its proposal early. This makes more likely that every other node will receive its proposal. That is, if the proposal takes 3.5s to be propagated, it will still be accepted, since the other processes are actually waiting 4s before prevoting nil. This is not the case of proposals produced by other nodes.

Reply:
> If blocks are accepted faster, the other proposers turns will also arrive earlier, correct, and therefore everybody will benefit?

True, they will benefit too. But Bob will still be the fastest producer

Reply:
I think that the problem here is having a node progressing faster than the others. The `timeout_commit` is the minimum duration of a height, do you agree?

If few nodes ignore this parameter, as described above, I don't really see a problem here. Except for the fact that they will not reward all the validators that helped deciding the previous block.

Reply:
An information here that I am pretty sure, is that a node that is waiting for `timeout_commit` will not prevote until it expires. So, yes, the proposal arrives faster but is not that everyone will move faster because the proposer doesn't wait before proposing at round `0` .

Reply:
I think someone mentioned here that there is no clear incentive to wait for `timeout_commit` and this is simply bad.

Reply:
> So, yes, the proposal arrives faster but is not that everyone will move faster because the proposer doesn't wait before proposing at round 0 .

won't other nodes decide on value faster in such case? i.e. when they receive proposal, they `X = (prevote, then precommit, then wait for timeout_commit)`. the `X` time doesn't change, but the rate of blocks changes

Reply:
> I think someone mentioned here that there is no clear incentive to wait for `timeout_commit` and this is simply bad.

fwiw, `The more precommits the proposer includes, the larger P is`, where `P` determines the proposer reward, see:
https://docs.cosmos.network/v0.45/modules/distribution/03_begin_block.html#reward-to-the-validators

Reply:
So if Alice was the proposer at `H` and Bob will be the proposer at `H+1`, Bob has an incentive to wait for more commits for `H`, commits that he will include in the proposal for `H+1`.